scispacy/candidate_generation.py

```diff
@@ -375,6 +376,9 @@ def create_tfidf_ann_index(
         The kb items to generate the index and vectors for.
 
     """
+    if not scipy_supports_sparse_float16():
```

Comment:
Would be good to provide some more info in the message here, e.g. saying what scipy/python versions are needed